lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -36,11 +37,12 @@ def collect(cls, block_structure):
     def transform_block_filters(self, usage_info, block_structure):
         if not settings.FEATURES.get('ENABLE_PROCTORED_EXAMS', False):
             return [block_structure.create_universal_filter()]
+        # this needs to change ^^^
 
-        def is_proctored_exam_for_user(block_key):
+        def has_milestones_for_user(block_key):
```

Comment:
@nasthagiri I left it internal because it makes use of several fields that aren't accessible from the usage key passed in in `block_structure.create_removal_filter(...)`. Is there a way to pass them in as additional parameters to `has_pending_milestones_for_user`?